bindings/c/test/apitester/TesterCorrectnessWorkload.cpp

```diff
@@ -125,6 +134,66 @@ class ApiCorrectnessWorkload : public ApiWorkload {
 		    });
 	}
 
+	void randomGetKeyOp(TTaskFct cont) {
+		int numKeys = Random::get().randomInt(1, maxKeysPerTransaction);
+		auto keysWithSelectors = std::make_shared<std::vector<std::pair<fdb::Key, fdb::KeySelector>>>();
+		auto selectors = std::make_shared<std::vector<fdb::KeySelector>>();
+		auto results = std::make_shared<std::vector<fdb::Key>>();
+		for (int i = 0; i < numKeys; i++) {
+			auto key = randomKey(readExistingKeysRatio);
+			fdb::KeySelector selector;
+			selector.orEqual = Random::get().randomBool(0.5);
+			selector.offset = Random::get().randomInt(0, 4);
+			keysWithSelectors->push_back({ key, selector });
+		}
+		execTransaction(
+		    [keysWithSelectors, results](auto ctx) {
+			    auto futures = std::make_shared<std::vector<fdb::Future>>();
+			    for (const auto& keyWithSelector : *keysWithSelectors) {
+				    auto key = keyWithSelector.first;
+				    auto selector = keyWithSelector.second;
+				    selector.key = key.data();
+				    selector.keyLength = key.size();
```

Comment:
Maybe fill out the KeySelector completely in one place (when filling out the keyWithSelectors) and just pass keyWithSelector.second to getKey. To make the code simpler

Reply:
You're right, done, although the code is still a bit tricky. Added a comment (also sort of explaining why I originally separated it.)